dsgrid/config/representative_period_time_dimension_config.py

```diff
@@ -196,6 +211,38 @@ def check_dataset_time_consistency(self, expected_timestamps, load_data_df, time
         DSGInvalidDataset
 
         """
+        logger.info("Check %s dataset time consistency.", self.__class__.__name__)
+        actual_timestamps = []
+        for row in load_data_df.select(*time_columns).distinct().sort(*time_columns).collect():
```

Comment:
Logic is moved, unchanged. Also, this could be pure spark and faster, but it likely doesn’t matter because the number of rows is small.